src/pages/ShowFeelingsPage/ShowFeelingTemplate.tsx

```diff
@@ -1,59 +1,70 @@
 /* eslint-disable react/prop-types */
-import React from 'react';
+import * as React from 'react';
 import {
   Button, Nav, Container,
 } from 'react-bootstrap';
 import { useRecoilValue } from 'recoil';
 import { useTranslation } from 'react-i18next';
 
 import { darkModeState } from '@store';
+import { IFeelingItem } from '@models';
+import { removeFeeling } from '@api/FeelingsAPI';
 
 import '@translations/i18n';
 import './ShowFeelingsPage.scss';
 
-interface IShowFeelingTemplateProp {
-  id: number;
-  category: string;
-  content: string;
-  date: Date;
+interface IProps {
+  feelingsListObject: IFeelingItem[],
+  setFeelingsListObject: (newFeelingsList: IFeelingItem[]) => void,
+  currentFeelingsList: IFeelingItem[],
 }
 
-export const ShowFeelingTemplate: React.FC<IShowFeelingTemplateProp> = ({ feelingsListObject }) => {
+export const ShowFeelingTemplate: React.FC<IProps> = ({ feelingsListObject, setFeelingsListObject, currentFeelingsList }) => {
   const { t } = useTranslation();
   const darkModeStatus = useRecoilValue(darkModeState);
   return (
     <div>
       <Container fluid>
         <div className="feelings-menu-desktop">
           {feelingsListObject && Object.keys(feelingsListObject)
-            .map(((feelingObject) => (
+            .map(((feelingId: string) => (
               <Button
-                key={feelingsListObject[feelingObject].content}
+                key={feelingsListObject[Number(feelingId)].content}
                 className={
                 darkModeStatus ? 'btn-my-feelings-dark btn-feelings-dark' : 'btn-my-feelings-light btn-feelings-light'
               }
                 size="lg"
-                onClick={() => console.log('TODO: Add remove feeling function')}
+                onClick={() => {
+                  removeFeeling(feelingsListObject[Number(feelingId)].id);
+                  let newFeelingsList = currentFeelingsList;
```

Comment:
Suggestion : In Showfeelings page, you already calling getfeelings in useeffect. Make a function of that in the same file. 
Use that here. This would reduce the lines of code because anyways you are close to O(n) complexity in both cases i guess.

Reply:
This is for modularization. One is the template and another is the populator so I don't think it matters anyway